File: sort-n8.py
def insertion_sort(arr):
    n = len(arr)
    for i in range(1, n):  # Start from the 2nd element
        key = arr[i]        # Element to insert
        j = i - 1
        # Shift elements > key to the right
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key    # Insert key in correct position
        print(f"Pass {i}: {arr}")  # Optional: Show each pass

# Example usage
arr = [29, 10, 14, 37, 13, 5, 8, 16]
print("Original Array:", arr)
insertion_sort(arr)
print("Sorted Array:", arr)

File: sort-n8.js
function insertionSort(arr) {
    const n = arr.length;
    
    for (let i = 1; i < n; i++) {
        // Current element to be inserted
        const key = arr[i];
        let j = i - 1;

        // Shift elements of arr[0..i-1] that are > key
        // to one position ahead of their current position
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key; // Insert key in its correct position
    }
    return arr; // Return the sorted array
}

// Example Usage
const array = [29, 10, 14, 37, 13, 5, 8, 16];
console.log("Original Array:", array);
const sortedArray = insertionSort(array);
console.log("Sorted Array:", sortedArray);